plugins/lsp/lsp-status.nix

```
@@ -0,0 +1,83 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+helpers.neovim-plugin.mkNeovimPlugin config {
+  name = "lsp-status";
+  originalName = "lsp-status.nvim";
+  defaultPackage = pkgs.vimPlugins.lsp-status-nvim;
+  maintainers = [ helpers.maintainers.b3nb5n ];
+  settingsOptions =
+    let
+      indicator_opt = helpers.defaultNullOpts.mkStr "" ''
+        string to show as diagnostics.
```

```suggestion
        The string to show as diagnostics.
```

If you wanted, you could do something along the lines of

```nix
mkIndicatorOpt = name: helpers.defaultNullOpts.mkStr "" ''
  The string to show for ${name} diagnostics.
'';
```

```nix
indicator_ok = mkIndicatorOpt "ok";
```

---

Actually, looking at your defaults test, you'll need to do something like that in order to correctly set the plugin-default for each option:

```nix
mkIndicatorOpt = default: name: helpers.defaultNullOpts.mkStr default ''
  The string to show for ${name} diagnostics.
'';
```

---

Marking unresolved because the second comment regarding plugin defaults wasn't addressed

---

the defaults should be unset because this option is used to override the defaults set by the plugin. is there a nullable string option that you use?

---

The "default" passed to `defaultNullOpts` functions is not the nix option default (that's always`null`), rather it is the "plugin default" which is used for our documentation only.

---

updated to use the nerdfont icons for the defaults